packages/@aws-cdk/aws-servicecatalog/lib/product.ts

```diff
@@ -0,0 +1,212 @@
+import { IResource, Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { AcceptLanguage } from './common';
+import { InputValidator } from './private/validation';
+import { CfnCloudFormationProduct } from './servicecatalog.generated';
+import { CloudFormationTemplate } from './template';
+
+/**
+ * A Service Catalog product, currently only supports type CloudFormationProduct
+ */
+export interface IProduct extends IResource {
+  /**
+   * The ARN of the product.
+   * @attribute
+   */
+  readonly productArn: string;
+
+  /**
+   * The id of the product
+   * @attribute
+   */
+  readonly productId: string;
+}
+
+abstract class ProductBase extends Resource implements IProduct {
+  public abstract readonly productArn: string;
+  public abstract readonly productId: string;
+}
+
+/**
+ * Properties of product version (also known as a provisioning artifact).
+ */
+export interface ProductVersion {
+  /**
+   * The description of the product version
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * Whether the specified product template will be validated by CloudFormation. If turned off, an invalid template configuration can be stored.
+   * @default true
+   */
+  readonly validateTemplate?: boolean;
+
+  /**
+   * The S3 template that points to the provisioning version template
+   */
+  readonly cloudFormationTemplate: CloudFormationTemplate;
+
+  /**
+   * The name of the product version.
+   * @default - No product version name provided
+   */
+  readonly name?: string;
+}
+
+/**
+ * Properties for a Cloudformation Product
+ */
+export interface CloudFormationProductProps {
+  /**
+   * The owner of the product.
+   */
+  readonly owner: string;
+
+  /**
+   * The name of the product.
+   */
+  readonly productName: string;
+
+  /**
+   * The configuration of the product version.
+   */
+  readonly productVersions: ProductVersion[];
+
+  /**
+   * The language code.
+   * @default - No accept language provided
+   */
+  readonly acceptLanguage?: AcceptLanguage;
+
+  /**
+   * The description of the product.
+   * @default - No description provided
+   */
+  readonly description?: string;
+
+  /**
+   * The distributor of the product.
+   * @default - No distributor provided
+   */
+  readonly distributor?: string;
+
+  /**
+   * Whether to give provisioning artifacts a new unique identifier when the product attributes or provisioning artifacts is updated
+   * @default false
+   */
+  readonly replaceProvisioningArtifacts?: boolean;
```

Comment:
Since you renamed the `provisioningArtifacts` property to `productVersions`, I would rename this to `replaceProductVersionIds`.